internal/clients/team3/client.go

```diff
@@ -1,34 +1,93 @@
-// Package team3 contains code for team 3's client implementation
-package team3
-
-import (
-	"fmt"
-	"log"
-
-	"github.com/SOMAS2020/SOMAS2020/internal/common"
-)
-
-const id = common.Team3
-
-func init() {
-	common.RegisterClient(id, &client{id: id})
-}
-
-type client struct {
-	id common.ClientID
-}
-
-func (c *client) Echo(s string) string {
-	c.Logf("Echo: '%v'", s)
-	return s
-}
-
-func (c *client) GetID() common.ClientID {
-	return c.id
-}
-
-// Logf is the client's logger that prepends logs with your ID. This makes
-// it easier to read logs. DO NOT use other loggers that will mess logs up!
-func (c *client) Logf(format string, a ...interface{}) {
-	log.Printf("[%v]: %v", c.id, fmt.Sprintf(format, a...))
-}
+// Package team3 contains code for team 3's client implementation
+package team3
+
+import (
+	"fmt"
+	"log"
+
+	"github.com/SOMAS2020/SOMAS2020/internal/common"
+	"github.com/SOMAS2020/SOMAS2020/internal/common/rules"
+	"gonum.org/v1/gonum/mat"
+)
+
+const id = common.Team3
+
+func init() {
+	common.RegisterClient(id, &client{id: id})
+}
+
+type client struct {
+	id common.ClientID
+}
+
+func BasicRuleEvaluator(ruleName string) bool {
+	rm := rules.AvailableRules[ruleName]
+
+	variables := rm.RequiredVariables
+
+	var variableVect []float64
+
+	for _, v := range variables {
+		variableVect = append(variableVect, rules.VariableMap[v]...)
+	}
+
+	variableVect = append(variableVect, 1)
+
+	variableFormalVect := mat.NewVecDense(len(variables)+1, variableVect)
+
+	rows, _ := rm.ApplicableMatrix.Dims()
+
+	actual := make([]float64, rows)
+
+	c := mat.NewVecDense(rows, actual)
+
+	c.MulVec(&rm.ApplicableMatrix, variableFormalVect)
+	aux := rm.AuxiliaryVector
+
+	var resultVect []bool
+
+	for i := 0; i < rows; i++ {
+		switch interpret := aux.AtVec(i); interpret {
+		case 0:
+			resultVect = append(resultVect, c.AtVec(i) == 0)
+		case 1:
+			resultVect = append(resultVect, c.AtVec(i) > 0)
+		case 2:
+			resultVect = append(resultVect, c.AtVec(i) >= 0)
+		case 3:
+			resultVect = append(resultVect, c.AtVec(i) != 0)
+		}
+	}
```

Comment:
should this logic be in client or common code?

Reply:
```golang
resultVect := []bool {
    c.AtVec(0) == 0,
    c.AtVec(1) > 0,
    ...
}
```

Reply:
So this is where we have to make a decision whether to give all clients a basic version of the Matrix evaluator in the common client file, or whether to tell them to implement it themselves using a provided spec. I've put it in a specific client for now because it's less disruptive to change this code while we are undecided on that.

Reply:
```golang
	for i := 0; i < rows; i++ {
                res := false
		switch interpret := aux.AtVec(i); interpret {
		case 0:
			res = c.AtVec(i) == 0
		...
                default:
                       // return an error? once this is factored out in a function
                 resultVect = append(resultVect, res)
	}
```

Reply:
Yup done